packages/math-input/src/components/keypad/mobile-keypad.tsx

```diff
@@ -0,0 +1,158 @@
+ * is responsible for connecting the keypad with MathInput and the Renderer.
+ *
+ * Ideally this strategy of attaching methods on the class component for
+ * other components to call will be replaced props/callbacks since React
+ * doesn't support this type of code anymore (functional components
+ * can't have methods attached to them).
+ */
+
+type Props = {
+    onElementMounted?: (arg1: any) => void;
+    onDismiss?: () => void;
+    style?: StyleType;
+};
+
+type State = {
+    active: boolean;
+    keypadConfig?: KeypadConfiguration;
+    keyHandler?: KeyHandler;
+    cursor?: Cursor;
+};
+
+class MobileKeypad extends React.Component<Props, State> {
+    hasMounted = false;
+
+    state: State = {
+        active: false,
+    };
+
+    activate: () => void = () => {
+        this.setState({active: true});
+    };
+
+    dismiss: () => void = () => {
+        this.setState({active: false});
+    };
+
+    configure: (configuration: KeypadConfiguration, cb: () => void) => void = (
+        configuration,
+        cb,
+    ) => {
+        this.setState({keypadConfig: configuration});
+
+        // HACK(charlie): In Perseus, triggering a focus causes the keypad to
+        // animate into view and re-configure. We'd like to provide the option
+        // to re-render the re-configured keypad before animating it into view,
+        // to avoid jank in the animation. As such, we support passing a
+        // callback into `configureKeypad`. However, implementing this properly
+        // would require middleware, etc., so we just hack it on with
+        // `setTimeout` for now.
+        setTimeout(() => cb && cb());
+    };
+
+    setCursor: (cursor: Cursor) => void = (cursor) => {
+        this.setState({cursor});
+    };
+
+    setKeyHandler: (keyHandler: KeyHandler) => void = (keyHandler) => {
+        this.setState({keyHandler});
+    };
+
+    getDOMNode: () => ReturnType<typeof ReactDOM.findDOMNode> = () => {
+        return ReactDOM.findDOMNode(this);
+    };
+
+    _handleClickKey(key: Key) {
+        if (key === "DISMISS") {
+            this.dismiss();
+            return;
+        }
+
+        const cursor = this.state.keyHandler?.(key);
+        this.setState({cursor});
+    }
+
+    render(): React.ReactNode {
+        const {active, cursor, keypadConfig} = this.state;
+
+        const containerStyle = [
+            styles.keypadContainer,
+            active ? styles.activeKeypadContainer : null,
+        ];
+
+        const isExpression = keypadConfig?.keypadType === "EXPRESSION";
+
+        return (
+            <View
+                style={containerStyle}
+                ref={(element) => {
+                    if (!this.hasMounted && element) {
+                        // Append the dispatch methods that we want to expose
+                        // externally to the returned React element.
+                        const elementWithDispatchMethods = {
+                            ...element,
```

Comment:
This made me pause.  We're spreading a DOM node into the object we're returning. This is even more unclear by the fact that the `onElementMounded` prop is typed with an `any` argument. 

So two things:

1) is it possible to not spread the DOM node into this object (keep the element as a pure API)?
2) can we add better typing to the `onElementMounted` prop?

Reply:
Totally fine with that! Thanks for creating the ticket.